syntax = "proto3";

package bili;

import "base.proto";

message MedalInfo {
    uint32 target_id = 1;
    int32 int2 = 2;
    string name = 3;

    // maybe
    int32 color = 4;
    int32 color_start = 5;
    int32 color_end = 6;
    int32 color_border = 7;

    uint32 roomid = 12;
    uint32 int4 = 13;
}

message UMedalInfo {
    string name = 1;
    uint32 level = 2;
    // maybe
    int32 color_start = 3;
    int32 color_end = 4;
    int32 color_border = 5;
    int32 color = 6;
    uint32 id = 7;
    uint32 ruid = 10;
    uint32 int4 = 12; // == MedalInfo.int4
    // maybe
    string v2_medal_color_start = 15;
    string v2_medal_color_end = 16;
    string v2_medal_color_border = 17;
    string v2_medal_text = 18;
    string v2_medal_level = 19;
}

message InteractWordUserInfo {
    message Message1 {
        uint32 int1 = 1;
    }
    uint32 uid = 1;
    UserBase base = 2;
    UMedalInfo medal_info = 3; // 可能同 指定用户的所有粉丝勋章信息 data.list[n].uinfo_medal
    Message1 message1 = 4;
    string string1 = 6;
}

// example:
//    https://i0.hdslb.com/bfs/live/bb88734558c6383a4cfb5fa16c9749d5290d95e8.png
//    曾经活跃过，近期与你互动较少
//    4
message ActivateMessage {
    string icon = 1;
    string msg = 2;
    int32 type = 3;
}

message InteractWordV2 {
    uint32 uid = 1;
    string uname = 2;
    string string1 = 4;
    uint32 msg_type = 5;
    uint32 roomid = 6;
    uint32 timestamp = 7;
    uint32 timestamp_millisecond = 8;
    MedalInfo medal_info = 9;
    string string2 = 12;
    int32 int2 = 15;
    int32 int3 = 17;
    string string4 = 19;
    InteractWordUserInfo user_info = 22;
    ActivateMessage activate_message = 23; // relation_tain in issue#1100
}
